src/test/java/it/eng/idsa/dataapp/util/HttpHeadersUtilTest.java

```diff
@@ -0,0 +1,124 @@
+		assertEquals(headers.get("IDS-Id").get(0), message.getId().toString());
+		assertEquals(headers.get("IDS-SecurityToken-TokenValue").get(0), message.getSecurityToken().getTokenValue());
+	}
+
+	@Test
+	public void messageToHeadersTest_ContractRequestMessage() {
+		Message message = UtilMessageService.getContractRequestMessage();
+
+		HttpHeaders headers = HttpHeadersUtil.messageToHttpHeaders(message);
+
+		assertNotNull(headers.entrySet());
+		assertEquals(headers.get("IDS-Messagetype").get(0), "ids:ContractRequestMessage");
+		assertEquals(headers.get("IDS-Id").get(0), message.getId().toString());
+		assertEquals(headers.get("IDS-SecurityToken-TokenValue").get(0), message.getSecurityToken().getTokenValue());
+	}
+
+	@Test
+	public void messageToHeadersTest_ContractAgreementMessage() {
+		Message message = UtilMessageService.getContractAgreementMessage();
+
+		HttpHeaders headers = HttpHeadersUtil.messageToHttpHeaders(message);
+
+		assertNotNull(headers.entrySet());
+		assertEquals(headers.get("IDS-Messagetype").get(0), "ids:ContractAgreementMessage");
+		assertEquals(headers.get("IDS-Id").get(0), message.getId().toString());
+		assertEquals(headers.get("IDS-SecurityToken-TokenValue").get(0), message.getSecurityToken().getTokenValue());
+	}
+
+	@Test
+	public void messageToHeadersTest_ArtifactRequestMessage() {
+		Message message = UtilMessageService.getArtifactRequestMessage();
+
+		HttpHeaders headers = HttpHeadersUtil.messageToHttpHeaders(message);
+
+		assertNotNull(headers.entrySet());
+		assertEquals(headers.get("IDS-Messagetype").get(0), "ids:ArtifactRequestMessage");
+		assertEquals(headers.get("IDS-Id").get(0), message.getId().toString());
+		assertEquals(headers.get("IDS-SecurityToken-TokenValue").get(0), message.getSecurityToken().getTokenValue());
+	}
+
+	@Test
+	public void headersToMessage_DescriptionRequestMessage() {
+		htppHeaders.add("ids-id",
+				"https://w3id.org/idsa/autogen/descriptionRequestMessage/eb3ab487-dfb0-4d18-b39a-585514dd044f");
+
+		Message httpHeadersToMessage = HttpHeadersUtil.httpHeadersToMessage(htppHeaders);
+
+		assertNotNull(httpHeadersToMessage);
+		assertEquals(htppHeaders.get("ids-id").get(0), httpHeadersToMessage.getId().toString());
+	}
+
+	@Test
+	public void headersToMessage_ContractRequestMessage() {
+		htppHeaders.add("ids-id",
+				"https://w3id.org/idsa/autogen/contractRequestMessage/eb3ab487-dfb0-4d18-b39a-585514dd044f");
+
+		Message httpHeadersToMessage = HttpHeadersUtil.httpHeadersToMessage(htppHeaders);
+
+		assertNotNull(httpHeadersToMessage);
+		assertEquals(htppHeaders.get("ids-id").get(0), httpHeadersToMessage.getId().toString());
+	}
+
+	@Test
+	public void headersToMessage_ContractAgreementMessage() {
+		htppHeaders.add("ids-id",
+				"https://w3id.org/idsa/autogen/contractAgreementMessage/eb3ab487-dfb0-4d18-b39a-585514dd044f");
+
+		Message httpHeadersToMessage = HttpHeadersUtil.httpHeadersToMessage(htppHeaders);
+
+		assertNotNull(httpHeadersToMessage);
```

Comment:
Can we check here if message (can you also rename it) is instance of ContractAgreementMessage, since test should check that, if naming is correct?
Maybe to add some more checks, if it is populated correct, token, date, issuerConnector, those mandatory fields.
Same for other "headers to message" tests.

Reply:
I enriched the test and added all checks that you mentioned.